### PR TITLE
Failed tests GraphQl query 

### PR DESCRIPTION
**Pull request by RulaKhaled**
### Purpose/Motivation
What is the feature? Why is this being done?
We need to query tests of repository query for the new test results tab. 
closes: https://github.com/codecov/engineering-team/issues/2069 

### What does this PR do?
Include a brief description of the changes in this PR. Bullet points are your friend.

### Notes to Reviewer
Anything to note to the team? Any tips on how to review, or where to start?

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: graphql_api/types/test_results/test_results.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from reports.models import Test
+
+test_result_bindable = ObjectType("TestResult")
+
+@test_result_bindable.field("name")
+def resolve_name(test: Test, info) -> str:
+    return test.name
+
+@test_result_bindable.field("updatedAt")
+def resolve_updated_at(test: Test, info) -> str:
```
Comment (JerrySentry) [inline graphql_api/types/test_results/test_results.py:12]: nit: `-> datetime`

**Review by JerrySentry — APPROVED** [review]

File: graphql_api/helpers/connection.py [hunk 2]
```diff
@@ -68,6 +68,38 @@ def page_info(self, *args, **kwargs):
         }
 
 
+class BetterCursorPaginator(CursorPaginator):
```
Comment (RulaKhaled) [inline graphql_api/helpers/connection.py:71]: Nice work.

Comment (ajay-sentry) [reply graphql_api/helpers/connection.py:71]: @joseph-sentry Can we update the name to DictCursorPaginator or something similarly less ambiguous

File: graphql_api/types/test_results/test_results.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+test_result_bindable = ObjectType("TestResult")
+
+
+@test_result_bindable.field("name")
+def resolve_name(test, info) -> str:
+    return test["test__name"]
+
+
+@test_result_bindable.field("updatedAt")
+def resolve_updated_at(test, info) -> str:
```
Comment (RulaKhaled) [inline graphql_api/types/test_results/test_results.py:14]: Fix datetime type

**Pull request merged**